j5/components/gpio_pin.py

```diff
@@ -0,0 +1,191 @@
+"""Classes for GPIO Pins."""
+
+from abc import abstractmethod
+from enum import IntEnum
+from typing import List, Optional, Type
+
+from j5.boards import Board
+from j5.components import Component, Interface, NotSupportedByHardware
+
+
+class BadGPIOPinMode(Exception):
+    """The pin is not in the correct mode."""
+
+    pass
+
+
+class GPIOPinMode(IntEnum):
+    """Hardware modes that a GPIO pin can be set to."""
+
+    DIGITAL_INPUT = 0  # The digital state of the pin can be read
```

Comment:
Changing these to `#:` will mean that Sphinx will use these as descriptors in the generated docs